owncloudApp/src/androidTest/java/com/owncloud/android/shares/ui/usecases/CreatePublicShareTest.kt

```diff
@@ -0,0 +1,373 @@
+        onView(withText(newPublicShare2.name)).check(matches(isDisplayed()))
+
+        /**
+         * 3rd public share
+         */
+        onView(withId(R.id.addPublicLinkButton)).perform(click())
+
+        val newPublicShare3 = publicShares[2]
+        savePublicShare(newPublicShare3)
+
+        // New share properly created
+        sharesLiveData.postValue(
+            Resource.success(
+                publicShares
+            )
+        )
+
+        // Check whether the dialog to create the public share has been properly closed
+        onView(withText(R.string.share_via_link_create_title)).check(doesNotExist())
+        onView(withText(newPublicShare3.name)).check(matches(isDisplayed()))
+    }
+
+    @Test
+    fun createShareLoading() {
+        loadCapabilitiesSuccessfully()
+        loadSharesSuccessfully(arrayListOf())
+
+        onView(withId(R.id.addPublicLinkButton)).perform(click())
+
+        savePublicShare(publicShares[0], Resource.loading())
+
+        onView(withText(R.string.common_loading)).check(matches(isDisplayed()))
+    }
+
+    @Test
+    fun createShareError() {
+        loadCapabilitiesSuccessfully()
+        loadSharesSuccessfully(arrayListOf())
+
+        onView(withId(R.id.addPublicLinkButton)).perform(click())
+
+        savePublicShare(
+            publicShares[0],
+            Resource.error(
+                RemoteOperationResult.ResultCode.FORBIDDEN,
+                exception = Exception("Error when retrieving shares")
+            )
+        )
+
+        onView(withText(R.string.share_link_file_error)).check(matches(isDisplayed()))
+    }
+
+    private fun getOCFileForTesting(name: String = "default") = OCFile("/Photos").apply {
+        availableOfflineStatus = OCFile.AvailableOfflineStatus.NOT_AVAILABLE_OFFLINE
+        fileName = name
+        fileId = 9456985479
+        remoteId = "1"
+        privateLink = "private link"
+    }
+
+    private fun loadCapabilitiesSuccessfully(
+        capability: OCCapability = TestUtil.createCapability(
+            versionString = "10.1.1",
+            sharingPublicMultiple = CapabilityBooleanType.TRUE.value
+        )
+    ) {
+        capabilitiesLiveData.postValue(
+            Resource.success(
+                capability
+            )
+        )
+    }
+
+    private fun loadSharesSuccessfully(shares: ArrayList<OCShare> = publicShares) {
+        sharesLiveData.postValue(Resource.success(shares))
+    }
+
+    private fun savePublicShare(newShare: OCShare, resource: Resource<Unit> = Resource.success()) {
+        `when`(
+            ocShareViewModel.insertPublicShareForFile(
+                1,
+                newShare.name!!,
+                "",
+                -1,
+                false
+            )
+        ).thenReturn(
+            MutableLiveData<Resource<Unit>>().apply {
+                postValue(resource)
+            }
+        )
+
+        onView(withId(R.id.saveButton)).perform(click())
+    }
+}
```

Comment:
Line at the end of the file, please

Reply:
The files do contain the line at the end of the file, thought GH does not show it. If the lines were not there, GH would show a red arrow.